### ISSUE TITLE
XCode iOS builds fail after using AdvancedSecurityCodeqlInit task

### ISSUE THREAD

**Issue by geertbleyen** [github/codeql-action#1899]
# Environment
Version of codeql-action: v2.14.6
Programming Language: Swift
Self-Hosting agent platform: MacOS 13.5.2
Build Target platform: iOS 16.2
Azude Devops Services: 

# Issue
After using the AdvancedSecurityCodeqlInit task in our pipeline, the subsequent task which builds our application for iOS (Using the xcode task in Azure Devops) no longer builds correctly.
From what I gather from the logs, after running the task AdvancedSecurityCodeqlInit, xcode no longer builds the correct platform and uses the wrong SDK and fails (uses MacOS platform and SDK, instead of iOS)
Running the same xcodebuild commandline from the terminal yields a successful build.
Running the same pipeline with the AdvancedSecurityCodeql tasks removed also yield a successful build.

# Logs
[AdvancedSecurityCodeqlInit.log](https://github.com/github/codeql-action/files/12698949/AdvancedSecurityCodeqlInit.log)
[xcodebuild.log](https://github.com/github/codeql-action/files/12698959/xcodebuild.log)

**Comment by AlexDenisov** [comment]
Hi @geertbleyen, looking at the logs it seems that the build is actually correct (modulo that it fails 😅): the macOS part is building Swift package descriptions. What fails is some weird mismatch between Swift toolchains while building those packages, e.g.:

```
// Slightly reformatted for readability
/Users/svcp_adsagtmotion/azagents/A1-mac/_work/2/s/Packages/SpeechRecognition/Package.swift:1:8: error: failed to build module 'Foundation'; 
this SDK is not supported by the compiler (the SDK is built with 'Apple Swift version 5.9 (swiftlang-5.9.0.123.203 clang-1500.0.37.1)',
while this compiler is 'Apple Swift version 5.9 (swiftlang-5.9.0.128.108 clang-1500.0.40.1)').
Please select a toolchain which matches the SDK.
```

CodeQL doesn't support Swift 5.9 yet so I'm not sure where the toolchain discrepancy is coming from.

**Comment by geertbleyen** [comment]
We are using XCode 15 RTM (forgot to mention in the environment info), which does use Swift 5.9 by default.
I understand what you mean with the MacOS platform/sdk bit, so yeah, it's not the crosscompile issue I presumed it would be. But I also don't know where these 2 different swift 5.9 toolchains would be coming from.

**Comment by charEtoile** [comment]
I have the same issue but two years later.

**Comment by rmakliuk** [comment]
@geertbleyen did you manage to bypass this?

**Comment by geertbleyen** [comment]
> [@geertbleyen](https://github.com/geertbleyen) did you manage to bypass this?

Nope, in the end we disabled the task and I moved on to another project.

**Comment by charEtoile** [comment]
Yes and no, I no longer use those fancy and often incomplete predefined tasks, I just use scripts.

**Comment by rmakliuk** [comment]
I’m not sure if this applies to your case, but I solved it by passing this environment variable to the `Xcode@5` task:

```yaml
- task: Xcode@5
  env:
    SEMMLE_PRELOAD_libtrace: $(SEMMLE_PRELOAD_LIBTRACE)
```
*Note*: SEMMLE_PRELOAD_LIBTRACE is set by AdvancedSecurity-Codeql-Init@1
### Environment

- **codeql-action version:** `v2.25.5`
- **Programming language:** `Swift`
- **Xcode version:** `26.2`
- **Microsoft-hosted agent:** `macos-15`
`
